products/surveys/backend/max_tools.py

```diff
@@ -0,0 +1,240 @@
+from posthog.models import Team
+from .prompts import SURVEY_CREATION_SYSTEM_PROMPT
+from .survey_schema import SurveyCreationOutput, DEFAULT_SURVEY_APPEARANCE
+
+
+class SurveyCreatorArgs(BaseModel):
+    instructions: str = Field(description="Natural language description of the survey to create")
+
+
+class SurveyCreatorTool(MaxTool):
+    name: str = "create_survey"
+    description: str = "Create and optionally launch a survey based on natural language instructions"
+    thinking_message: str = "Creating your survey"
+
+    root_system_prompt_template: str = """
+    You are helping create surveys for this PostHog team.
+
+    Current context:
+    - Total surveys: {{{total_surveys_count}}}
+    - Recent surveys: {{{existing_surveys}}}
+
+    When creating surveys, consider the existing surveys to avoid duplication and suggest complementary survey strategies.
+    """
+
+    args_schema: type[BaseModel] = SurveyCreatorArgs
+
+    def _run_impl(self, instructions: str) -> tuple[str, dict[str, Any]]:
+        """
+        Generate survey configuration from natural language instructions.
+        """
+        try:
+            # Get team for context
+            team = Team.objects.get(id=self._team_id)
+
+            # Create the prompt
+            prompt = ChatPromptTemplate.from_messages(
+                [
+                    ("system", SURVEY_CREATION_SYSTEM_PROMPT),
+                    ("human", "Create a survey based on these instructions: {{{instructions}}}"),
+                ],
+                template_format="mustache",
+            )
+
+            # Set up the LLM with structured output
+            model = (
+                ChatOpenAI(model="gpt-4.1-mini", temperature=0.2)
+                .with_structured_output(SurveyCreationOutput, include_raw=False)
+                .with_retry()
+            )
+
+            # Generate the survey configuration
+            chain = prompt | model
+            result = chain.invoke(
+                {
+                    "instructions": instructions,
+                    "existing_surveys": self._get_existing_surveys_summary(),
+                    "team_survey_config": self._get_team_survey_config(team),
+                }
+            )
+
+            # Convert to PostHog survey format and create directly
+            survey_data = self._convert_to_posthog_format(result, team)
+
+            # Use the proper serializer for validation and creation
+            from posthog.api.survey import SurveySerializerCreateUpdateOnly
+            from datetime import datetime
+
+            # Set launch date if requested
+            if result.should_launch:
+                survey_data["start_date"] = datetime.now()
+
+            # Get the actual user from the context
+            user = None
+            user_id = self.context.get("user_id")
+            if not user_id:
+                return "❌ Failed to create survey: User authentication required", {"error": "authentication_required"}
```

Comment:
see https://github.com/PostHog/posthog/pull/33111/files#r2154356156
we should not even call the backend if theres no user

Reply:
agreed. I still left the validation though, but changed the error and moved it to the beginning of the function to validate if it's present